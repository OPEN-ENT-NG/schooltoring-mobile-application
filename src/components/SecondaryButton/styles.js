import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  button: {
    padding: 10,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  },
  buttonText: {
    fontSize: 18,
    color: COLORS.PRIMARY
  }
});
