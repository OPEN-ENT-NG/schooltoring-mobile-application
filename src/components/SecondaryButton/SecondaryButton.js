import React from "react";
import { Text, View } from "react-native";
import PropTypes from "prop-types";

import Touchable from "../Touchable/Touchable";

import styles from "./styles";
import { COLORS } from "../../styles/common";

const SecondaryButton = props => {
  const children = props.children || (
    <Text style={styles.buttonText}>{props.title}</Text>
  );

  return (
    <View
      style={[
        props.style || {},
        styles.button,
        props.disabled && { backgroundColor: COLORS.GREY }
      ]}
    >
      <Touchable
        disabled={props.disabled}
        onPress={() => props.onPress()}
        hitSlop={{
          top: 10,
          left: 10,
          right: 10,
          bottom: 10
        }}
      >
        {children}
      </Touchable>
    </View>
  );
};

SecondaryButton.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  style: PropTypes.any,
  title: PropTypes.string,
  disabled: PropTypes.bool
};

export default SecondaryButton;
