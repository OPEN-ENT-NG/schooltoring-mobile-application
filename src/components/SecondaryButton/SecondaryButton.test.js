import React from "react";
import { TouchableOpacity, Text } from "react-native";
import { shallow } from "enzyme";

import { SecondaryButton } from "./SecondaryButton";

describe("Render", () => {
  test("SecondaryButton should render without crashing", () => {
    const component = shallow(
      <SecondaryButton onPress={jest.fn()} title="Test" />
    );
    expect(component).toMatchSnapshot();
  });
  test("An error should be logged if no onPress or title given", () => {
    const PropTypes = jest.spyOn(global.console, "error");
    const rendered = shallow(<SecondaryButton />);
    expect(rendered).toMatchSnapshot();
    expect(PropTypes).toHaveBeenCalled();
  });
  test("Rendered title is the same than the title prop given", () => {
    const title = "Button title";
    const rendered = shallow(<SecondaryButton title={title} />);

    expect(
      rendered
        .find(Text)
        .first()
        .contains(title)
    );
  });
});

describe("Interactions", () => {
  test("Press on the back button should trigger mocked function", () => {
    const mockCallBack = jest.fn();
    const component = shallow(
      <SecondaryButton onPress={mockCallBack} title="Test" />
    );
    component.find(TouchableOpacity).prop("onPress")();
    expect(mockCallBack).toHaveBeenCalled();
  });
});
