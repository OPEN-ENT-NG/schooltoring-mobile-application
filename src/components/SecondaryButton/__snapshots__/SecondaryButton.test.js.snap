// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render An error should be logged if no onPress or title given 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onPress={[Function]}
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#17b6b0",
        "fontSize": 18,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Render SecondaryButton should render without crashing 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onPress={[Function]}
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#17b6b0",
        "fontSize": 18,
      }
    }
  >
    Test
  </Text>
</TouchableOpacity>
`;
